demos/python_demos/im2latex_demo/im2latex_demo.py

```diff
@@ -0,0 +1,207 @@
+from __future__ import print_function
+
+import datetime
+import logging as log
+import os
+import sys
+from argparse import SUPPRESS, ArgumentParser
+from collections import namedtuple
+from copy import deepcopy
+from time import time
+
+import cv2
+import numpy as np
+from tqdm import tqdm
+
+import os,sys,inspect
+
+from utils import BatchCropPadToTGTShape, BatchResizePadToTGTShape
+from utils import END_TOKEN, START_TOKEN, read_vocab
+from openvino.inference_engine import IECore, IENetwork
+
+COLOR_WHITE = 255
+
+PREPROCESSING = {
+    'BatchCropPadToTGTShape': BatchCropPadToTGTShape,
+    'BatchResizePadToTGTShape': BatchResizePadToTGTShape
+}
+
+def print_stats(module):
+    perf_counts = module.requests[0].get_perf_counts()
+    print('{:<70} {:<15} {:<15} {:<15} {:<10}'.format('name', 'layer_type', 'exet_type', 'status',
+                                                      'real_time, us'))
+    for layer, stats in perf_counts.items():
+        print('{:<70} {:<15} {:<15} {:<15} {:<10}'.format(layer, stats['layer_type'], stats['exec_type'],
+                                                          stats['status'], stats['real_time']))
+
+
+def preprocess_image(preprocess, image_raw, tgt_shape):
+    target_height, target_width = tgt_shape
+    image_raw = preprocess(tgt_shape)(image_raw)[0]
+    assert image_raw.shape[0] == target_height and image_raw.shape[1] == target_width, image_raw.shape
+    image = image_raw / float(COLOR_WHITE)
+    image = image.transpose((2, 0, 1))
+    return image
+
+
+def build_argparser():
+    parser = ArgumentParser(add_help=False)
+    args = parser.add_argument_group('Options')
+    args.add_argument('-h', '--help', action='help',
+                      default=SUPPRESS, help='Show this help message and exit.')
+    args.add_argument("-m_encoder", help="Required. Path to an .xml file with a trained encoder part of the model",
+                      required=True, type=str)
+    args.add_argument("-m_dec_step", help="Required. Path to an .xml file with a trained decoder step part of the model",
+                      required=True, type=str)
+    args.add_argument("-i", "--input", help="Required. Path to a folder with images or path to an image files",
+                      required=True, type=str, nargs="+")
+    args.add_argument("-o", "--output_file",
+                      help="Optional. Path to file where to store output. If not mentioned, result will be stored"
+                      "in the console.",
+                      type=str)
+    args.add_argument("--vocab_path", help="Path to vocab file to construct meaningful phrase",
+                      type=str)
+    args.add_argument("--target_shape", help="Required. Target image shape (height, width). "
```

Comment:
It was previously planned to support reshape of the network according to the shape of the input image, that is the reason, why network is loaded in the loop on the input images. But it looks like this release there is no support of the reshape of this concrete network, so you are right, it would be better to use shape of the input layer of the network.